### PR TITLE
SP forms release: "Understand SharePoint form integration" article

### PR DESCRIPTION
**Pull request by sarafankit**
Share/Anne - Please review this and we should release this when we release other SharePoint Form documentation. Can you please also include a link to this article in the customize-sharepoint-form as well? I could not find that article in the repository.

### CODE REVIEW

**Comment by mgblythe** [comment]
Closing and reopening to trigger content validation / staging links

File: powerapps-docs/sharepoint-form-integration.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+Now that you have a good understanding of the default generated form and the **SharePointIntegration** control, you can change the formulas to customize the forms further. Here are few tips on how to customize the forms:
+
+* To customize the experience separately for creating, showing, or editing an item; set the **OnNew**, **OnView**, or **OnEdit** formulas of **SharePointIntegration** control to set variables or navigate to different screens.
+
+* Use the **OnSave** formula of **SharePointIntegration** control to customize what happens when user clicks or taps **Save** in SharePoint. If you have multiple forms, make sure to submit the changes only for the form currently being used.
+
+    * **Tip**: Set different values for a variable in **OnNew**, **OnView**, and **OnEdit** formulas. You can use this variable in **OnSave** formula to determine which form is being used.
+
+* Make sure to include **RequestHide()** in **OnSuccess** of all your forms. If you forget this, SharePoint will not know when to hide the form.
```
Comment (SKjerland) [inline powerapps-docs/sharepoint-form-integration.md]: Make sure to include **RequestHide()** in the **OnSuccess** formula for all your forms. If you forget this, SharePoint will not know when to hide the form.

File: powerapps-docs/sharepoint-form-integration.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+* You cannot control hiding of the form when user clicks or taps **Cancel** in SharePoint. So make sure to reset your forms in **OnCancel** formula of **SharePointIntegration** control.
```
Comment (SKjerland) [inline powerapps-docs/sharepoint-form-integration.md]: You can't control the hiding of a form when a user clicks or taps **Cancel** in SharePoint, so make sure you reset your forms in the **OnCancel** formula of the **SharePointIntegration** control.

File: powerapps-docs/connections/connection-sharepoint-online.md [hunk 3]
```diff
@@ -30,36 +30,22 @@ You can add data from a custom list but not a library. In addition, not all type
 | --- | --- | --- |
 | Single line of text |Yes |View text |
 | Multiple lines of text |Yes |View text |
-| Choice |Yes (single values only) |View lookup |
+| Choice |Yes |View lookup<br>Edit lookup |
```
[the remark this answers is not in the document]
Comment (sarafankit) [reply powerapps-docs/connections/connection-sharepoint-online.md]: Done

File: powerapps-docs/connections/connection-sharepoint-online.md [hunk 4]
```diff
@@ -30,36 +30,22 @@ You can add data from a custom list but not a library. In addition, not all type
...
 | Number |Yes |View percentage<br>View rating<br>View text |
 | Currency |Yes |View percentage<br>View rating<br>View text |
 | Date and Time |Yes |View text |
-| Lookup |Yes (single values only) |View lookup<br>Edit lookup |
+| Lookup |Yes |View lookup<br>Edit lookup |
```
[the remark this answers is not in the document]
Comment (sarafankit) [reply powerapps-docs/connections/connection-sharepoint-online.md]: Done

File: powerapps-docs/connections/connection-sharepoint-online.md [hunk 5]
```diff
@@ -30,36 +30,22 @@ You can add data from a custom list but not a library. In addition, not all type
...
 | Boolean (Yes/No) |Yes |View text<br>View toggle |
-| Person or Group |Yes (single values only) |View lookup<br>Edit lookup |
+| Person or Group |Yes |View lookup<br>Edit lookup |
```
[the remark this answers is not in the document]
Comment (sarafankit) [reply powerapps-docs/connections/connection-sharepoint-online.md]: Done

File: powerapps-docs/sharepoint-form-integration.md [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+ms.devlang: na
+ms.topic: article
+ms.tgt_pltfrm: na
+ms.workload: na
+ms.date: 11/11/2017
+ms.author: ankitsar
+
+---
+# Understand SharePoint forms integration
+You can now easily [customize any SharePoint list form](customize-sharepoint-form.md) in PowerApps. In this article, we will walk through the details of how these forms work and how you can customize them further.
```
[the remark this answers is not in the document]
Comment (sarafankit) [reply powerapps-docs/sharepoint-form-integration.md]: Done

File: powerapps-docs/sharepoint-form-integration.md [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+
+If you have customized a form for a SharePoint list, you must have experienced that the default generated form just works for all operations like creating, showing or editing an item. This is accomplished by the help of generated formulas and the SharePointIntegration control.
```
[the remark this answers is not in the document]
Comment (sarafankit) [reply powerapps-docs/sharepoint-form-integration.md]: Done

File: powerapps-docs/sharepoint-form-integration.md [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+
+## Understand the default generated form
+
+The default generated form consists of the following controls and their corresponding defaults:
+
+* **FormScreen1** - This is the [screen](./controls/control-screen.md) that contains the form.
+
+* **SharePointForm1** - This is the [form](working-with-forms.md) that is used to create, show or edit the list item.
```
[the remark this answers is not in the document]
Comment (sarafankit) [reply powerapps-docs/sharepoint-form-integration.md]: Done

File: powerapps-docs/sharepoint-form-integration.md [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+
+    * **Data Source** - The list for which the form has been customized.
+
+    * **Item** - The selected item from list. We also set this to First() of the list for your convenience when working in PowerApps Studio.
```
[the remark this answers is not in the document]
Comment (sarafankit) [reply powerapps-docs/sharepoint-form-integration.md]: Done

File: powerapps-docs/sharepoint-form-integration.md [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+
+        **If(IsBlank(SharePointIntegration.Selected) || IsEmpty(SharePointIntegration.Selected),First('*YourListName*'),SharePointIntegration.Selected)**
+
+    * **OnSuccess** -  Once the item is created or saved successfully, the form is reset and SharePoint is requested to hide the form.
```
[the remark this answers is not in the document]
Comment (sarafankit) [reply powerapps-docs/sharepoint-form-integration.md]: Done

File: powerapps-docs/sharepoint-form-integration.md [hunk 11]
```diff
@@ -0,0 +1,111 @@
...
+
+        **ResetForm(SharePointForm1); RequestHide()**
+
+* **SharePointIntegration** - The control responsible for communicating user actions between SharePoint and PowerApps.
+
+    * **Data Source** - The list for which the form has been customized.
+
+        **'*YourListName*'**
+
+    * **OnNew** - Set the **SharePointForm1** in new mode.
```
[the remark this answers is not in the document]
Comment (sarafankit) [reply powerapps-docs/sharepoint-form-integration.md]: Done

File: powerapps-docs/sharepoint-form-integration.md [hunk 12]
```diff
@@ -0,0 +1,111 @@
...
+
+        **NewForm(SharePointForm1)**
+
+    * **OnView** - Set the **SharePointForm1** in view mode.
```
[the remark this answers is not in the document]
Comment (sarafankit) [reply powerapps-docs/sharepoint-form-integration.md]: Done

File: powerapps-docs/sharepoint-form-integration.md [hunk 13]
```diff
@@ -0,0 +1,111 @@
...
+
+        **ViewForm(SharePointForm1)**
+
+    * **OnEdit** - Set the **SharePointForm1** in edit mode.
```
[the remark this answers is not in the document]
Comment (sarafankit) [reply powerapps-docs/sharepoint-form-integration.md]: Done

File: powerapps-docs/sharepoint-form-integration.md [hunk 14]
```diff
@@ -0,0 +1,111 @@
...
+
+        **EditForm(SharePointForm1)**
+
+    * **OnSave** - Submit the changes to **SharePointForm1**. On successful submission of the form, **SharePointForm1.OnSuccess** formula will be executed.
```
[the remark this answers is not in the document]
Comment (sarafankit) [reply powerapps-docs/sharepoint-form-integration.md]: Done

File: powerapps-docs/sharepoint-form-integration.md [hunk 15]
```diff
@@ -0,0 +1,111 @@
...
+
+        **SubmitForm(SharePointForm1)**
+
+    * **OnCancel** - Reset the changes to **SharePointForm1**. SharePoint always hides the form when user clicks or taps cancel button in SharePoint.
```
[the remark this answers is not in the document]
Comment (sarafankit) [reply powerapps-docs/sharepoint-form-integration.md]: Done

File: powerapps-docs/sharepoint-form-integration.md [hunk 16]
```diff
@@ -0,0 +1,111 @@
...
+
+        **SubmitForm(SharePointForm1)**
+
+These defaults ensure that the form works when running within SharePoint. They change the PowerApps form mode as the user interacts with in SharePoint and also ensure the changes are submitted to SharePoint.
```
[the remark this answers is not in the document]
Comment (sarafankit) [reply powerapps-docs/sharepoint-form-integration.md]: Done

File: powerapps-docs/sharepoint-form-integration.md [hunk 17]
```diff
@@ -0,0 +1,111 @@
...
+
+## Understand the SharePointIntegration control
+The **SharePointIntegration** control is responsible for communicating user actions between SharePoint and PowerApps.
```
[the remark this answers is not in the document]
Comment (sarafankit) [reply powerapps-docs/sharepoint-form-integration.md]: Done

File: powerapps-docs/sharepoint-form-integration.md [hunk 18]
```diff
@@ -0,0 +1,111 @@
...
+
+![](./media/sharepoint-form-integration/sharepointintegration-object.png)
+
+The properties for SharePointIntegration are only available when the form is running in SharePoint and cannot be accessed when customizing the form in PowerApps studio.
```
[the remark this answers is not in the document]
Comment (sarafankit) [reply powerapps-docs/sharepoint-form-integration.md]: Done

[the remark this answers is not in the document]
Comment (sarafankit) [reply powerapps-docs/sharepoint-form-integration.md]: Changed this to Note as well

File: powerapps-docs/sharepoint-form-integration.md [hunk 19]
```diff
@@ -0,0 +1,111 @@
...
+
+It has the following properties:
```
[the remark this answers is not in the document]
Comment (sarafankit) [reply powerapps-docs/sharepoint-form-integration.md]: Done

File: powerapps-docs/sharepoint-form-integration.md [hunk 20]
```diff
@@ -0,0 +1,111 @@
...
+
+**Selected** - The selected item from the SharePoint list.
+
+**OnNew** - How an app responds when user clicks or taps **New** button or opens the create item form in SharePoint.
```
[the remark this answers is not in the document]
Comment (sarafankit) [reply powerapps-docs/sharepoint-form-integration.md]: Done

File: powerapps-docs/sharepoint-form-integration.md [hunk 21]
```diff
@@ -0,0 +1,111 @@
...
+
+**OnView** - How an app responds when user clicks or taps an **item** or opens the detail form in SharePoint.
```
[the remark this answers is not in the document]
Comment (sarafankit) [reply powerapps-docs/sharepoint-form-integration.md]: Done

File: powerapps-docs/sharepoint-form-integration.md [hunk 22]
```diff
@@ -0,0 +1,111 @@
...
+
+**OnEdit** - How an app responds when user clicks or taps **Edit all** button or opens the edit item form in SharePoint.
```
[the remark this answers is not in the document]
Comment (sarafankit) [reply powerapps-docs/sharepoint-form-integration.md]: Done

File: powerapps-docs/sharepoint-form-integration.md [hunk 23]
```diff
@@ -0,0 +1,111 @@
...
+
+**OnSave** - How an app responds when user clicks or taps **Save** button in SharePoint.
```
[the remark this answers is not in the document]
Comment (sarafankit) [reply powerapps-docs/sharepoint-form-integration.md]: Done

File: powerapps-docs/sharepoint-form-integration.md [hunk 24]
```diff
@@ -0,0 +1,111 @@
...
+
+**OnCancel** - How an app responds when user clicks or taps **Cancel** button in SharePoint.
```
[the remark this answers is not in the document]
Comment (sarafankit) [reply powerapps-docs/sharepoint-form-integration.md]: Done

File: powerapps-docs/sharepoint-form-integration.md [hunk 25]
```diff
@@ -0,0 +1,111 @@
...
+
+**SelectedListItemID** - Item id for the selected item in SharePoint list.
```
[the remark this answers is not in the document]
Comment (sarafankit) [reply powerapps-docs/sharepoint-form-integration.md]: Done

File: powerapps-docs/sharepoint-form-integration.md [hunk 26]
```diff
@@ -0,0 +1,111 @@
...
+
+**Data Source** - The list that contains the record that the form will show, edit, or create.
+
+* If you change this property, the **Selected** and **SelectedItemID** property may stop working.
```
[the remark this answers is not in the document]
Comment (sarafankit) [reply powerapps-docs/sharepoint-form-integration.md]: Done

File: powerapps-docs/sharepoint-form-integration.md [hunk 27]
```diff
@@ -0,0 +1,111 @@
...
+
+## Customize the default form
+
+Now that you have a good understanding of the default generated form and the **SharePointIntegration** control, you can change the formulas to customize the forms further. Here are few tips on how to customize the forms:
+
+* To customize the experience separately for creating, showing, or editing an item; set the **OnNew**, **OnView**, or **OnEdit** formulas of **SharePointIntegration** control to set variables or navigate to different screens.
+
+* Use the **OnSave** formula of **SharePointIntegration** control to customize what happens when user clicks or taps **Save** in SharePoint. If you have multiple forms, make sure to submit the changes only for the form currently being used.
```
[the remark this answers is not in the document]
Comment (sarafankit) [reply powerapps-docs/sharepoint-form-integration.md]: Done

File: powerapps-docs/sharepoint-form-integration.md (hunk 2 above)
Comment (sarafankit) [reply powerapps-docs/sharepoint-form-integration.md]: Done

File: powerapps-docs/sharepoint-form-integration.md (hunk 1 above)
Comment (sarafankit) [reply powerapps-docs/sharepoint-form-integration.md]: Done

File: powerapps-docs/sharepoint-form-integration.md [hunk 28]
```diff
@@ -0,0 +1,111 @@
...
+
+    * **Tip**: Set different values for a variable in **OnNew**, **OnView**, and **OnEdit** formulas. You can use this variable in **OnSave** formula to determine which form is being used.
```
[the remark this answers is not in the document]
Comment (sarafankit) [reply powerapps-docs/sharepoint-form-integration.md]: Done

File: powerapps-docs/sharepoint-form-integration.md [hunk 29]
```diff
@@ -0,0 +1,111 @@
...
+
+**Data Source** - The list that contains the record that the form will show, edit, or create.
+
+* If you change this property, the **Selected** and **SelectedItemID** property may stop working.
+
+## Customize the default form
+
+Now that you have a good understanding of the default generated form and the **SharePointIntegration** control, you can change the formulas to customize the forms further. Here are few tips on how to customize the forms:
+
+* To customize the experience separately for creating, showing, or editing an item; set the **OnNew**, **OnView**, or **OnEdit** formulas of **SharePointIntegration** control to set variables or navigate to different screens.
```
[the remark this answers is not in the document]
Comment (sarafankit) [reply powerapps-docs/sharepoint-form-integration.md]: Done

**Comment by sarafankit** [comment]
#sign-off

**Pull request merged**